pkg/snapstore/snapshot.go

```diff
@@ -21,68 +21,97 @@ import (
 
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
 	"github.com/sirupsen/logrus"
 )
 
 // NewSnapshot returns the snapshot object.
-func NewSnapshot(kind string, startRevision, lastRevision int64, compressionSuffix string) *Snapshot {
-	snap := &Snapshot{
+func NewSnapshot(kind string, startRevision, lastRevision int64, compressionSuffix string) *brtypes.Snapshot {
+	snap := &brtypes.Snapshot{
 		Kind:              kind,
 		StartRevision:     startRevision,
 		LastRevision:      lastRevision,
 		CreatedOn:         time.Now().UTC(),
 		CompressionSuffix: compressionSuffix,
 	}
-	snap.GenerateSnapshotDirectory()
+	// snap.GenerateSnapshotDirectory()
 	snap.GenerateSnapshotName()
 	return snap
 }
 
-// GenerateSnapshotName prepares the snapshot name from metadata
-func (s *Snapshot) GenerateSnapshotName() {
-	s.SnapName = fmt.Sprintf("%s-%08d-%08d-%d%s", s.Kind, s.StartRevision, s.LastRevision, s.CreatedOn.Unix(), s.CompressionSuffix)
-}
+// ParseSnapshot parse <snapPath> to create snapshot structure
+func ParseSnapshot(snapPath string) (*brtypes.Snapshot, error) {
+	logrus.Infof("Snap path: %s", snapPath)
+	var err error
+	var backupVersion string = ""
+	s := &brtypes.Snapshot{}
+	// First try if the path contains v1
+	lastIndex := strings.LastIndex(snapPath, "v1/")
+	if lastIndex >= 0 {
+		backupVersion = backupVersionV1
+	}
 
-// GenerateSnapshotDirectory prepares the snapshot directory name from metadata
-func (s *Snapshot) GenerateSnapshotDirectory() {
-	s.SnapDir = fmt.Sprintf("Backup-%d", s.CreatedOn.Unix())
-}
+	// Then try if the path contains v2
+	if backupVersion == "" {
+		lastIndex = strings.LastIndex(snapPath, "v2/")
+		if lastIndex >= 0 {
+			backupVersion = backupVersionV2
+		}
+	}
 
-// GetSnapshotDirectoryCreationTimeInUnix returns the creation time for snapshot directory.
-func (s *Snapshot) GetSnapshotDirectoryCreationTimeInUnix() (int64, error) {
-	tok := strings.TrimPrefix(s.SnapDir, "Backup-")
-	return strconv.ParseInt(tok, 10, 64)
-}
+	if backupVersion == "" {
+		return nil, fmt.Errorf("invalid snapPath %v, does not contain any backupVersion", snapPath)
+	}
+
+	lastIndex = lastIndex + 3
+	prefix := snapPath[:lastIndex]
+	snapPath = snapPath[lastIndex:]
 
-// ParseSnapshot parse <snapPath> to create snapshot structure
-func ParseSnapshot(snapPath string) (*Snapshot, error) {
-	var err error
-	s := &Snapshot{}
 	tok := strings.Split(snapPath, "/")
-	logrus.Debugf("no of tokens:=%d", len(tok))
-	if len(tok) <= 1 || len(tok) > 3 {
+	if len(tok) < 1 || len(tok) > 3 {
 		return nil, fmt.Errorf("invalid snapshot name: %s", snapPath)
 	}
-	snapName := tok[1]
-	snapDir := tok[0]
+
+	var snapName, snapDir string = "", ""
+	// Get snap name from the tokens
+	// Consider the token before snap name
+	// If it's v1, then consider the token as snapDir
+	// If it's v2, then no snapDir exist
+	switch backupVersion {
+	case backupVersionV1:
+		snapName = tok[1]
+		snapDir = tok[0]
+	case backupVersionV2:
+		snapName = tok[0]
```

Comment:
```suggestion
	case backupVersionV1:
		snapName = path.Join(tok[1:]...)
		snapDir = tok[0]
	case backupVersionV2:
		snapName = path.Join(tok[0:]...)
```
To cover, deeper folder structures if any.

Reply:
We are taking care of that in later part.

Reply:
Done